src/biokbase/narrative/app_util.py

```diff
@@ -874,34 +884,43 @@ def transform_object_value(transform_type: Optional[str], value: Optional[str])
 
     if is_upa and transform_type in ["upa", "resolved-ref"]:
         return value
-    if is_ref and not is_upa and transform_type in ["ref", "unresolved-ref"]:
+    if (
+        not is_upa
+        and is_ref
+        and transform_type in ["ref", "unresolved-ref", "putative-ref"]
+    ):
         return value
     if not is_upa and not is_ref and transform_type is None:
         return value
 
-
     search_ref = value
     if not is_upa and not is_ref:
         search_ref = f"{system_variable('workspace')}/{value}"
     try:
         obj_info = clients.get("workspace").get_object_info3(
             {"objects": [{"ref": search_ref}]}
         )
-    except Exception:
+    except Exception as e:
+        # a putative-ref can be an extant or a to-be-created object; if the object
+        # is not found, the workspace name/object name can be returned
+        if transform_type == "putative-ref" and (
+            "No object with name" in str(e) or "No object with id" in str(e)
+        ):
+            return search_ref
+
         transform = transform_type
         if transform is None:
             transform = "object name"
         raise ValueError(
-            f"Unable to find object reference '{search_ref}' to transform as {transform}"
+            f"Unable to find object reference '{search_ref}' to transform as {transform}: "
+            + str(e)
         )
 
-    if is_path:
+    if is_path or transform_type in ["resolved-ref", "upa"]:
         return ";".join(obj_info["paths"][0])
-    if transform_type in ["ref", "unresolved-ref"]:
+    if transform_type in ["ref", "unresolved-ref", "putative-ref"]:
```

Comment:
`ref` and `unresolved-ref` do the same thing -- it would be good to get rid of `unresolved-ref` to make the specs less complicated.